ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -522,28 +527,27 @@ azure_file_read(
         std::cerr << "azure_file_read: read_object request to Azure failed." << std::endl;
         return ret;
     }
-    WT_UNUSED(session);
 
     return 0;
 }
 
-// Get the size of a file in bytes, by file handle.
+// Get the size of a file in bytes.
 static int
 azure_file_size(WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t *sizep)
 {
     azure_file_handle *azure_fh = reinterpret_cast<azure_file_handle *>(file_handle);
     int ret;
     bool exists;
+    WT_UNUSED(exists);
     size_t size = 0;
     if ((ret = azure_fh->fs->azure_conn->object_exists(azure_fh->name, exists, size)) != 0) {
         std::cerr << "azure_file_open: object_exists request to Azure failed." << std::endl;
         return ret;
     }
-    if (exists == false) {
-        std::cerr << "azure_file_open: no such file." << std::endl;
-        return EINVAL;
-    }
-    return size;
+
+    *sizep = size;
```

Comment:
I'm guessing this was your python test issue? If so, well spotted! One nitpick here, can we set `*sizep` to 0 at the start? That way even if someone ignores the return value they're less likely to do something insane like try to read a terabyte of garbage data because something was left uninitialised.

Reply:
thanks for the suggestion, resolved

Reply:
Thanks, this looks right after fixing the pointer thing.